core/sql/common/csconvert.cpp

```diff
@@ -1275,3 +1279,41 @@ char * findStartOfChar( char *someByteInChar, char *startOfBuffer )
      rtnv-- ;
   return rtnv ;
 }
+/* A method to do character set conversion , using Glibc iconv */
+int code_convert(const char *from_charset,const char *to_charset,char *inbuf, size_t inlen, char *outbuf,size_t outlen)
+{
+  iconv_t cd;
+  int rc;
+  char **pin = &inbuf;
+  char **pout = &outbuf;
+
+  cd = iconv_open(to_charset,from_charset);
+  if (cd==0) return -1;
+  memset(outbuf,0,outlen);
+  if (iconv(cd,pin,(size_t*)&inlen,pout,(size_t *)&outlen)==-1) 
+  {
+    iconv_close(cd);
+    return -1;
+  }
+  iconv_close(cd);
+  return outlen;
+}
+/* from gbk to utf8 */
+int gbk2utf8(char *inbuf,size_t inlen,char *outbuf,size_t outlen)
+{
+  return code_convert("gbk","utf-8",inbuf,inlen,outbuf,outlen);
+}
+
+int gbkToUtf8(char* gbkString, size_t gbklen, 
+              char* result ,size_t outlen, int addNullAtEnd)
```

Comment:
This 'addNullAtEnd is defined as int for all other similar functions, so following the convertion. But you are right, I will try to change it to bool.